crypto/async/async_wait.c

```diff
@@ -145,6 +145,7 @@ int ASYNC_WAIT_CTX_clear_fd(ASYNC_WAIT_CTX *ctx, const void *key)
     while (curr != NULL) {
         if (curr->del == 1) {
             /* This one has been marked deleted already so do nothing */
+            prev = curr;
             curr = curr->next;
             continue;
         }
```

Comment:
maybe remove everything including the continue and insert an else here?

Reply:
That's a good point to rais, but it seems like it would end up actually looking like

```
if (curr->del != 1 && curr->key == key) {
[...]
```

which doesn't leave as good a place for a comment explaining why we do nothing.
So I think I'm inclined to take Kurt's review and merge this as-is.

Reply:
I thought of
```
 if (curr->del == 1) {
   /* comment */
 }
 else if (curr->key == key) {
 }
 prev=curr;
 curr = curr->next;
```

but good enough as is.

Reply:
Ah, I guess I'm too used to worrying about compilers warning about empty blocks to have thought of that :)
Thanks for commenting.